services/121-service/src/payments/transactions/transaction.repository.ts

```diff
@@ -35,6 +36,57 @@ export class TransactionScopedRepository extends ScopedRepository<TransactionEnt
     return await query.getRawMany<GetAuditedTransactionDto>(); // Leaving this as getRawMany for now, as it is not a plain entity. It's a concatenation of multiple entities.
   }
 
+  public async getTransactionsForPayment({
```

Comment:
This function should not be in the Transactions Module, but in the Payments Module. Reason: the Transactions Module cannot know about higher-level concepts like registrations, and such.